pkg/dao/db.go

```diff
@@ -1,20 +1,55 @@
 package dao
 
-import "database/sql"
+import (
+	"database/sql"
+	"errors"
+	"fmt"
 
-var db *sql.DB
+	_ "github.com/lib/pq"
+)
 
-func Connect() (*sql.DB, error) {
-	if db != nil {
-		return db, nil
+type DatabaseHandler struct {
+	DB *sql.DB
+}
+
+func (handler *DatabaseHandler) Connect(dbHost, dbUser, dbPassword, dbName, sslMode string) error {
+	if handler.DB != nil {
+		return nil
 	}
 	//connection logic
-	return nil, nil
+	connStr := fmt.Sprintf("host=%v user=%v password=%v dbname=%v sslmode=%v", dbHost, dbUser, dbPassword, dbName, sslMode)
+
+	// sql.Open doesn't initialize the connection immediately
+	dbInstance, err := sql.Open("postgres", connStr)
+
+	// an error can happen here if the connection string is invalid
+	if err != nil {
+		return err
+	}
+
+	// an error happens here if we cannot connect
+	if err = dbInstance.Ping(); err != nil {
+		return err
+	}
+
+	// assign db variable declared above
```

Comment:
not sure we need that level of docs on the code 😄

Reply:
This is actually an outdated comment from when the db was declared at a package level. Good catch!